### ISSUE TITLE
Loss computed in numpy cause performance degradation

### ISSUE THREAD

**Issue by ThomasDelteil** [zackchase/mxnet-the-straight-dope#455]
In a lot of example notebooks, you use:

`curr_loss = nd.mean(loss).asscalar()`

which is the equivalent of `curr_loss = nd.mean(loss).assnumpy()[0]`

https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html#mxnet.ndarray.NDArray.asscalar

My tests gives 2-10% performance improvements when doing the .asscalar() on the moving loss at the end of the epoch rather than in every batch.

Related to this issue: https://github.com/apache/incubator-mxnet/issues/9571 which you opened @zackchase  😄
